src/economy/shipping_schedule.cc

```diff
@@ -0,0 +1,1715 @@
+			sslog("%s is idle\n", pair.first.get(game)->get_shipname().c_str());
+		} else if (update_me && update_me->serial() == pair.first.serial()) {
+			sslog("Recalculate for %s\n", pair.first.get(game)->get_shipname().c_str());
+			Path path;
+			pair.first.get(game)->calculate_sea_route(
+			   game, *pair.second.front().dock.get(game), &path);
+			int32_t d = -1;
+			game.map().calc_cost(path, &d, nullptr);
+			assert(d >= 0);
+			pair.second.front().duration_from_previous_location = d;
+		} else {
+			if (pair.second.front().duration_from_previous_location > time_since_last_update) {
+				pair.second.front().duration_from_previous_location -= time_since_last_update;
+				sslog("Regular-type heuristic update for %s\n",
+				      pair.first.get(game)->get_shipname().c_str());
+			} else {
+				// She said five more seconds, and that was ten seconds ago…
+				// The ship is behind schedule, so this is an arbitrary estimate
+				// about the arrival time. Doesn't matter if it's inaccurate,
+				// the ship will most likely arrive within a few seconds.
+				pair.second.front().duration_from_previous_location /= 2;
+				sslog("UNEXPECTED-type heuristic update for %s\n",
+				      pair.first.get(game)->get_shipname().c_str());
+			}
+		}
+	}
+	last_updated_ = time;
+
+	/* SECOND PASS:
+	 * Scan all ports. Make lists of waiting items.
+	 * Figure out when the items will be picked up.
+	 * Also cancel orders where we provided more capacity than is actually needed (which
+	 * can happen when a transfer is cancelled when the item is still in the portdock),
+	 * and cancel expedition ships in spe whose expeditions were cancelled.
+	 */
+	const size_t nr_ports = fleet_.get_ports().size();
+	sslog("SECOND PASS: %" PRIuS " ports\n", nr_ports);
+
+#ifndef NDEBUG
+	for (const auto& plan : plans_) {
+		assert(plan.second.size() <= nr_ports);
+	}
+#endif
+
+	if (nr_ports == 0) {
+		// Nothing to do. Ships stay where they are, or do whatever they want.
+		return endless();
+	}
+	std::list<Ship*> ships_with_reduced_orders;
+	std::list<PortDock*> ports_with_unserviced_expeditions;
+	// Don't even think about trying to cache any of this. It is impossible to maintain.
+	std::map<
+	   OPtr<PortDock> /* start */,
+	   std::map<OPtr<PortDock> /* dest */,
+	            std::pair<std::map<OPtr<Ship> /* by whom */,
+	                               std::pair<Duration /* when at `start` */,
+	                                         Quantity /* accept how much from `start` to `dest` */>>,
+	                      int32_t /* capacity missing (-) or extra (+) */>>>
+	   items_in_ports;
+	for (PortDock* dock : fleet_.get_ports()) {
+		const bool expedition_ready = dock->is_expedition_ready();
+		sslog("Iteration: dock %u (expedition ready %s)\n", dock->serial(),
+		      expedition_ready ? "true" : "false");
+		Ship* expedition_ship_coming = nullptr;
+		std::map<OPtr<PortDock>,
+		         std::pair<std::map<OPtr<Ship>, std::pair<Duration, Quantity>>, int32_t>>
+		   map;
```

Comment:
Better? :
		std::map<OPtr<PortDock>, /* src */
		         std::pair<std::map<OPtr<Ship>, /* Destination */
                            std::pair<Duration, Quantity>>, int32_t>>
		   port_planning_map;

Reply:
The automated formatting enforces a poorly readable format :(